### PR TITLE
Admin customers

### PR DESCRIPTION
**Pull request by nima-vova**

### CODE REVIEW

File: src/AppBundle/Admin/UserAdmin.php [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    protected function configureFormFields(FormMapper $formMapper)
+    {
+        $formMapper
+            ->with('User', ['class' => 'col-lg-12'])
+            ->add('firstName')
+            ->add('lastName')
+             ->add('email')
+
+            ->add('facebookId')
+            ->add('apiKey', 'text', ['attr' => ['readonly' => true]])
```
Comment (spolischook) [inline src/AppBundle/Admin/UserAdmin.php]: remove this. We will have "impersonate" functionality, but apiKey should be secured

File: src/AppBundle/Repository/SwindlerRepository.php [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<?php
+
+namespace AppBundle\Repository;
+
+class SwindlerRepository extends \Doctrine\ORM\EntityRepository
+{
+    public function findSwindlerIsBanned($ip)
```
Comment (spolischook) [inline src/AppBundle/Repository/SwindlerRepository.php]: bad method signature. 
First of all you should add docblock or return type.
If method return boolean method should start with ```is``` prefix, e.g. isIpBanish
If method return Entity|null method should start with ```find``` without ```is``` in method name

File: src/AppBundle/Controller/SwindlerController.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+use Symfony\Component\HttpFoundation\RedirectResponse;
+
+class SwindlerController extends CRUDController
+{
+    /**
+     * @return \Symfony\Component\HttpFoundation\RedirectResponse
+     */
+    public function lockAction()
+    {
+        $performanceEventId = $this->get('request')->query->get('id');
```
Comment (spolischook) [inline src/AppBundle/Controller/SwindlerController.php]: $performanceEventId - copy paste?
use param convertor for this

File: web/app/logs/111.log [hunk 4]
```diff
@@ -0,0 +1 @@
+[2017-04-11 05:59:09] invalid_api_key.INFO: Info message for interesting things [] []
```
Comment (spolischook) [inline web/app/logs/111.log]: remove this

File: src/AppBundle/Security/ApiKeyAuthenticator.php [hunk 5]
```diff
@@ -55,52 +67,81 @@ public function getUser($credentials, UserProviderInterface $userProvider)
...
      */
     public function supportsRememberMe()
     {
         return false;
     }
+
+    /**
+     * {@inheritdoc}
+     */
+    private function saveSwindler($request)
```
Comment (spolischook) [inline src/AppBundle/Security/ApiKeyAuthenticator.php]: too many ifs

File: src/AppBundle/Entity/Swindler.php [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+
+use Doctrine\ORM\Mapping as ORM;
+
+/**
+ * Swindler.
+ *
+ * @ORM\Table()
+ * @ORM\Entity(repositoryClass="AppBundle\Repository\SwindlerRepository")
+ */
+class Swindler
```
Comment (spolischook) [inline src/AppBundle/Entity/Swindler.php]: Not all clients are Swindlers. Lets rename this entity

File: src/AppBundle/Controller/SwindlerController.php (hunk 3 above)
Comment (nima-vova) [inline src/AppBundle/Controller/SwindlerController.php]: yes, $performanceEventId - copy paste. 
Maybe instead param convertor use 
$this->admin
            ->getSubject()
           ->setBanned(true);
        $em = $this->getDoctrine()->getManager();
        $em->flush($swindler);

Comment (spolischook) [inline src/AppBundle/Controller/SwindlerController.php]: What is $this->admin?

Comment (nima-vova) [inline src/AppBundle/Controller/SwindlerController.php]: SwindlerController extends CRUDController 
(use Sonata\AdminBundle\Controller\CRUDController;)
 admin:
https://github.com/sonata-project/SonataAdminBundle/blob/3.x/Admin/AdminInterface.php

File: app/config/config.yml [hunk 7]
```diff
@@ -67,12 +67,17 @@ doctrine:
         types:
             json: Sonata\Doctrine\Types\JsonType
             uuid_binary: Ramsey\Uuid\Doctrine\UuidBinaryType
+        mapping_types:
```
Comment (spolischook) [inline app/config/config.yml]: add only types that you really need. Add please comment to this config why you use it

File: app/config/parameters.yml.dist [hunk 8]
```diff
@@ -7,7 +7,7 @@ parameters:
     database_user:     root
     database_password: ~
     # You should uncomment this if you want use pdo_sqlite
-    # database_path: "%kernel.root_dir%/data.db3"
+    #database_path: "%kernel.root_dir%/data.db3"
```
Comment (spolischook) [inline app/config/parameters.yml.dist]: remove this, we are using mysql

File: app/config/services.yml [hunk 9]
```diff
@@ -66,3 +68,7 @@ services:
         tags:
             - { name: kernel.event_listener, event: kernel.exception }
 
+    customer_app.monolog_web_processor:
+         class: Monolog\Processor\WebProcessor
+         tags:
+            - { name: monolog.processor, channel: security_error }
```
Comment (spolischook) [inline app/config/services.yml]: image

File: src/AppBundle/Security/ApiKeyAuthenticator.php [hunk 10]
```diff
@@ -11,27 +13,37 @@
...
+        $this->logger = $logger;
     }
 
     /**
-     * @inheritdoc
+     * {@inheritdoc}
      */
     public function getCredentials(Request $request)
     {
+        $client = $this->registry->getRepository('AppBundle:Client')
```
Comment (spolischook) [inline src/AppBundle/Security/ApiKeyAuthenticator.php]: you are violate Single responsibility principle. Use separate event listener with high priority to hit this condition.

File: app/config/config.yml (hunk 7 above)
Comment (nima-vova) [inline app/config/config.yml]: Sorry, already removed. There was an old bug "Unknown database type enum requested, Doctrine\DBAL\Platforms\MySqlPlatform may not support it ".  It helped.

File: src/AppBundle/Security/ApiKeyAuthenticator.php (hunk 10 above)
Comment (nima-vova) [inline src/AppBundle/Security/ApiKeyAuthenticator.php]: Thank you. that's cool

**Comment by spolischook** [comment]
Please add functional test and remove extra log

File: src/AppBundle/Entity/Translations/ClientTranslation.php [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+
+/**
+ * @ORM\Entity
+ * @ORM\Table(name="client_translation",
+ *     uniqueConstraints={@ORM\UniqueConstraint(name="lookup_unique_role_translation_idx", columns={
+ *         "locale", "object_id", "field"
+ *     })}
+ * )
+ */
+class ClientTranslation extends AbstractPersonalTranslation
```
Comment (spolischook) [inline src/AppBundle/Entity/Translations/ClientTranslation.php]: we don't need translation for client. 
Even more, we don't need create/edit form for it

**Comment by spolischook** [comment]
We don't run fixtures on update build, you should load it in setUp method.

File: src/AppBundle/Entity/Translations/ClientTranslation.php (hunk 11 above)
Comment (nima-vova) [inline src/AppBundle/Entity/Translations/ClientTranslation.php]: Translation only used $ this-> processDeleteAction ($ object) in the test. Oк, delete his.

Comment (spolischook) [inline src/AppBundle/Entity/Translations/ClientTranslation.php]: Why we need translation for delete action? Maybe issue in other place?

Comment (nima-vova) [inline src/AppBundle/Entity/Translations/ClientTranslation.php]: Sorry, I was wrong, he did not like Softdeleteable. And Softdeleteable in entity not added, so erroneously did translation.  Need to clear the table.

**Comment by spolischook** [comment]
@nima-vova  thanks for your contribution! 
Hope you are well!
Unfortunately we decide use third party service for sell tickets, so I close this PR

**Pull request closed without merging**